src/ekat/ekat_assert.hpp

```diff
@@ -29,32 +29,32 @@
     }                                                                   \
   } while(0)
 
-#define impl_kthrow(condition, msg)             \
+#define IMPL_KERNEL_THROW(condition, msg)             \
   do {                                          \
     if ( ! (condition) )                        \
       Kokkos::abort(#condition "\n" msg);       \
   } while (0)
 
 #ifndef NDEBUG
-#define ekat_assert(condition)                impl_throw(condition, "",  std::logic_error)
-#define ekat_assert_msg(condition, msg)       impl_throw(condition, msg, std::logic_error)
-#define ekat_kassert(condition)               impl_kthrow(condition, "")
-#define ekat_kassert_msg(condition, msg)      impl_kthrow(condition, msg)
+#define EKAT_ASSERT(condition)                      IMPL_THROW(condition, "",  std::logic_error)
```

Comment:
What's the benefit of making all these names all caps?

Reply:
When you see `ekat_assert_msg(cond, "blah");` you are not sure if it is a fcn or a macro. All caps are almost a universal sign of either a macro or a compile-time constant. I think it's beneficial to let the user know that these are macros, not function.

This is an unlikely scenario for these macros, but still: taking the address of a fcn is ok, but if you try the same with a macro you might get a cryptic compilation errror.